### PR TITLE
Document deletion in the ingestor

### LINKED ISSUE #304: Document ingestors need a way to identify documents deleted at their source that also need to be deleted locally
## Proposal

- We need to add a mechanism to compare what we have in the local database to what we have at the remote ingestor source and keep them in sync. 
#### Indigo Adapter
- At the point where gather the `points in time` urls for updating, we can add a check for existing related works using the Work model and compare them with the list of urls in the point in time object. 
- If we have existing documents for the Work that are not in the point in time object, we can delete them

#### Gazettes Adapter

- For gazettes, we can do a straight comparison for values which are in the current database that do not exist in the Gazettes.africa database for the current jurisdiction. We delete the ones that we have in the database that do not exist in Gazettes.africa database.

### PR DESCRIPTION
**Pull request by actlikewill**
- Adds document deletion in the ingestor.

closes #304

### CODE REVIEW

**Comment by actlikewill** [comment]
@longhotsummer Turns out we already had a document deletion method in the adapter that we used with the indigo webhook

https://github.com/laws-africa/peachjam/blob/337e81a0367c449b4c8a115820a723db53cd4504/peachjam/adapters/adapters.py#L352-L365

File: peachjam/adapters/adapters.py
Comment (longhotsummer) [inline peachjam/adapters/adapters.py:106]: Just remember this must be scoped to only the place (country + locality) that the ingestor is applicable to. eg. AfricanLII we have four separate ingestors (EAC, ECOWAS, SADC, AU). They should only delete "their" legislation.

**Comment by longhotsummer** [comment]
I think we should update the Indigo adapter so we give it an API URL (eg. `https://api.laws.africa/v2` and a place code and it builds the full URL when it needs it. That way we don't need to do the weird `re.sub` to delete the document.

So, the updated and delete calls both get an expression FRBR URI, and the adapter generates a full url with `api_url + expression_frbr_uri`. The call to `get_doc_list()` would build up `api_url + '/' + place_code + '.json'`

File: peachjam/adapters/adapters.py
Comment (actlikewill) [reply peachjam/adapters/adapters.py]: I've added the jurisdiction and locality as settings which we can use here.

File: peachjam/adapters/adapters.py
Comment (longhotsummer) [inline peachjam/adapters/adapters.py:114]: if there is no locality, you must explicitly filter on `locality=None`.

File: peachjam/adapters/gazettes.py
Comment (longhotsummer) [inline peachjam/adapters/gazettes.py:53]: is the single underscore in `locality_code` correct?

File: peachjam/adapters/adapters.py
Comment (actlikewill) [reply peachjam/adapters/adapters.py:114]: Added. :heavy_check_mark:

File: peachjam/adapters/gazettes.py
Comment (actlikewill) [reply peachjam/adapters/gazettes.py]: Added :heavy_check_mark:

File: peachjam/adapters/adapters.py
Comment (longhotsummer) [reply peachjam/adapters/adapters.py:114]: did you git push?

Comment (actlikewill) [reply peachjam/adapters/adapters.py:114]: sorry, check now

**Review by longhotsummer — APPROVED** [review]
Cool, let's try it. Hopefully it doesn't delete everything accidentally!

**Comment by longhotsummer** [comment]
> > Cool, let's try it. Hopefully it doesn't delete everything accidentally!
> 
> lol now you've made me nervous 😅

Anything that automatically could potentially delete everything in a database should definitely make you nervous!

**Comment by actlikewill** [comment]
> > > Cool, let's try it. Hopefully it doesn't delete everything accidentally!
> > 
> > 
> > lol now you've made me nervous sweat_smile
> 
> Anything that automatically could potentially delete everything in a database should definitely make you nervous!

right. let me just double check everything before merging

**Comment by actlikewill** [comment]
@longhotsummer I want to wait for the pdf issue on indigo to resolve to merge this in. I want to test it properly.

**Comment by longhotsummer** [comment]
ok. The PDF issue is resolved.

**Pull request merged**
